packages/react-components/src/components/AppFrame/components/NavigationItem/NavigationItem.tsx

```diff
@@ -87,6 +89,7 @@ export const NavigationItem: React.FC<INavigationItemProps> = ({
             {...props}
           >
             {icon}
+            {isMobile && label}
```

Comment:
Instead of using separate prop isMobile, I'd use only the label and show it when this component placed inside of `MobileNavigation`.